### PR TITLE
[Promotion] Rework rule checkers for count & total to be more flexible

### PR DESCRIPTION
**Pull request by stloyd**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| License | MIT |

### CODE REVIEW

**Comment by stloyd** [comment]
Something missing @pjedrzejewski @michalmarcinkowski ?

**Comment by pjedrzejewski** [comment]
Equal with values ``equal``, ``more_than`` etc. does not sound right ... We will need different comparators in many other rules, so would be good to standardize this right now.

**Comment by stloyd** [comment]
I know naming is not best, but still better than `true/false` :wink:

**Comment by stloyd** [comment]
Also I was thinking about simple trait for such comparsion, which would unify other places too.

**Comment by stloyd** [comment]
@pjedrzejewski WDYT?

**Comment by stloyd** [comment]
@pjedrzejewski @michalmarcinkowski ping pong poke... :)

File: src/Sylius/Bundle/PromotionBundle/Form/Type/Rule/ItemCountConfigurationType.php [hunk 1]
```diff
@@ -34,11 +34,13 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     new Type(array('type' => 'numeric')),
                 ),
             ))
-            ->add('equal', 'checkbox', array(
-                'label' => 'sylius.form.rule.item_count_configuration.equal',
-                'constraints' => array(
-                    new Type(array('type' => 'bool')),
-                ),
+            ->add('equal', 'choice', array(
```
Comment (lchrusciel) [inline src/Sylius/Bundle/PromotionBundle/Form/Type/Rule/ItemCountConfigurationType.php]: Please be consistent ;) Once you use `array(` (see below) once you put `[` :)

File: src/Sylius/Component/Promotion/Checker/CountComparisonTrait.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Promotion\Checker;
+
+trait CountComparisonTrait
+{
+    /**
+     * @param int    $count
```
Comment (lchrusciel) [inline src/Sylius/Component/Promotion/Checker/CountComparisonTrait.php]: We are not aligning params vertically any more.

**Comment by pjedrzejewski** [comment]
@stloyd We do not want to touch these atm as I am not entirely sure about this solution and we will be doing some work based on #3368.

**Comment by stloyd** [comment]
@pjedrzejewski Please wait a week or less with approach based on #3368, I have done something almost identical for company I work now & must talk about open-sourcing the code (we use only API so there is no frontend for it unfortunatelly ;)), and as well this code is one of the changes done for that approach (just extracted).

**Pull request closed without merging**